lms/djangoapps/instructor/views/api.py

```diff
@@ -2821,21 +2821,50 @@ def start_certificate_regeneration(request, course_id):
     """
     course_key = CourseKey.from_string(course_id)
     certificates_statuses = request.POST.getlist('certificate_statuses', [])
-    if not certificates_statuses:
+
+    verified_with_audit_certs = False
+    if 'verified_users_with_audit_certs' in certificates_statuses:
```

Comment:
I see why this is tricky: we're regenerating here on a calculated value, rather than a stored value. What's wrong with putting `verified_users_with_audit_certs` into `allowed_statuses`? That way you avoid this pop/index business in the next couple of lines